singularity-eos/eos/modifiers/floored_energy.hpp

```diff
@@ -0,0 +1,374 @@
+//------------------------------------------------------------------------------
+// © 2021-2024. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+//------------------------------------------------------------------------------
+
+#ifndef _SINGULARITY_EOS_EOS_FLOORED_ENERGY_
+#define _SINGULARITY_EOS_EOS_FLOORED_ENERGY_
+
+#include "stdio.h"
+#include <cstdlib>
+#include <iostream>
+
+#include <ports-of-call/portability.hpp>
+#include <singularity-eos/base/eos_error.hpp>
+#include <singularity-eos/eos/eos_base.hpp>
+
+namespace singularity {
+
+using namespace eos_base;
+
+template <typename T>
+class FlooredEnergy : public EosBase<FlooredEnergy<T>> {
+ public:
+  // Generic functions provided by the base class. These contain
+  // e.g. the vector overloads that use the scalar versions declared
+  // here We explicitly list, rather than using the macro because we
+  // overload some methods.
+  using EosBase<FlooredEnergy<T>>::TemperatureFromDensityInternalEnergy;
+  using EosBase<FlooredEnergy<T>>::InternalEnergyFromDensityTemperature;
+  using EosBase<FlooredEnergy<T>>::PressureFromDensityTemperature;
+  using EosBase<FlooredEnergy<T>>::PressureFromDensityInternalEnergy;
+  using EosBase<FlooredEnergy<T>>::MinInternalEnergyFromDensity;
+  using EosBase<FlooredEnergy<T>>::EntropyFromDensityTemperature;
+  using EosBase<FlooredEnergy<T>>::EntropyFromDensityInternalEnergy;
+  using EosBase<FlooredEnergy<T>>::SpecificHeatFromDensityTemperature;
+  using EosBase<FlooredEnergy<T>>::SpecificHeatFromDensityInternalEnergy;
+  using EosBase<FlooredEnergy<T>>::BulkModulusFromDensityTemperature;
+  using EosBase<FlooredEnergy<T>>::BulkModulusFromDensityInternalEnergy;
+  using EosBase<FlooredEnergy<T>>::GruneisenParamFromDensityTemperature;
+  using EosBase<FlooredEnergy<T>>::GruneisenParamFromDensityInternalEnergy;
+  using EosBase<FlooredEnergy<T>>::FillEos;
```

Comment:
There are now some more general macros that can replace these lines defined in `eos_base.hpp`

Reply:
Yes! I'll change that

Reply:
Fixed